### PR TITLE
feat: use gmp sdk utils

### PR DESCRIPTION
**Pull request by Foivos**
[AXΕ-2415](https://axelarnetwork.atlassian.net/browse/AXE-2415)

### CODE REVIEW

File: contracts/interchain-token-service/InterchainTokenService.sol [hunk 1]
```diff
@@ -52,7 +52,7 @@ contract InterchainTokenService is
     address internal immutable implementationLockUnlockFee;
     IAxelarGateway public immutable gateway;
     IAxelarGasService public immutable gasService;
-    IRemoteAddressValidator public immutable remoteAddressValidator;
+    IInterchainRouter public immutable interchainRouter;
```
Comment (milapsheth) [inline contracts/interchain-token-service/InterchainTokenService.sol:55]: also need to update ITS deploy script

File: contracts/interchain-token-service/InterchainTokenService.sol [hunk 2]
```diff
@@ -678,7 +679,8 @@ contract InterchainTokenService is
      * @param gasValue The amount of gas to be paid for the transaction
      */
     function _callContract(string calldata destinationChain, bytes memory payload, uint256 gasValue) internal {
-        string memory destinationAddress = remoteAddressValidator.getRemoteAddress(destinationChain);
+        string memory destinationAddress = interchainRouter.getTrustedAddress(destinationChain);
+        if(bytes(destinationAddress).length == 0) revert UntrustedChain(destinationChain);
```
Comment (milapsheth) [inline contracts/interchain-token-service/InterchainTokenService.sol:683]:
```suggestion
        if (bytes(destinationAddress).length == 0) revert UntrustedChain(destinationChain);

```
should just revert in `getTrustedAddress` I think

File: contracts/utils/Imports.sol [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { InterchainRouter } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/utils/InterchainRouter.sol';
+import { InterchainRouterProxy } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/utils/InterchainRouterProxy.sol';
```
Comment (milapsheth) [inline contracts/utils/Imports.sol:6]: once gmp sdk is released, we can skip these imports?

Comment (Foivos) [reply contracts/utils/Imports.sol]: no, they are needed for tests (hardhat won't know they exists unless they are built by `hardhat compile` and you need to import them somewhere.

File: contracts/interchain-token-service/InterchainTokenService.sol (hunk 1 above)
Comment (Foivos) [reply contracts/interchain-token-service/InterchainTokenService.sol]: done

File: contracts/utils/Imports.sol [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { InterchainAddressTracker } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/utils/InterchainAddressTracker.sol';
```
Comment (milapsheth) [inline contracts/utils/Imports.sol:5]: if it's only for tests, could consider moving this to test folder

File: contracts/utils/Imports.sol [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+import { InterchainAddressTrackerProxy } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/utils/InterchainAddressTrackerProxy.sol';
```
Comment (milapsheth) [inline contracts/utils/Imports.sol:6]: ignore warnings

**Review by milapsheth — APPROVED** [review]

File: DESIGN.md [hunk 6]
```diff
@@ -32,7 +32,7 @@ Most projects that look to go cross-chain nowadays have more complex needs that
 
 ## Linker Router
 
-We plan to finalize the design of the `InterchainTokenService` but we want to be able to support new chains as they get added to the Axelar Network. For this purpose, the service will ask a separate contract, the [`RemoteAddressValidator`](./contracts/remoteAddressValidator/RemoteAddressValidator.sol) to obtain the destination address for outgoing messages, and for validation of incoming messages. This contract might eventually stop being upgradable but it will probably be able to support new addresses for new chains indefinitely.
+We plan to finalize the design of the `InterchainTokenService` but we want to be able to support new chains as they get added to the Axelar Network. For this purpose, the service will ask a separate contract, the [`InterchainAddressTracker`](./contracts/interchainAddressTracker/InterchainAddressTracker.sol) to obtain the destination address for outgoing messages, and for validation of incoming messages. This contract might eventually stop being upgradable but it will probably be able to support new addresses for new chains indefinitely.
```
[the remark this answers is not in the document]
Comment (Foivos) [reply DESIGN.md:35]: fixed

File: contracts/utils/Imports.sol (hunk 4 above)
Comment (Foivos) [reply contracts/utils/Imports.sol:5]: sure

File: contracts/utils/Imports.sol (hunk 5 above)
Comment (re1ro) [reply contracts/utils/Imports.sol]: @Foivos @milapsheth why do we need this proxy at all? We should just use a regular Proxy.

**Review by re1ro — APPROVED** [review]

**Review by milapsheth — APPROVED** [review]

**Pull request merged**